prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt

```diff
@@ -635,6 +636,56 @@ hnm8COa8Kr+bnTqzScpQuOfujHcFEtfcYUGfSS6HusxidwXx+lYi1A==
         assertThat(retryItems).isNull()
     }
 
+    @Test
+    fun `test  transport postReport with valid file name for Natus`() {
+        val header = makeHeader()
+        val mockRestTransport = spyk(RESTTransport(mockClientStringTokenOk()))
+
+        // Given:
+        //      lookupDefaultCredential returns mock UserPassCredential object to allow
+        //      the getAuthTokenWithUserPass() to be called.
```

Comment:
Why are you copying the filename logic here? You shouldn't have to copy the same logic you're testing in the unit test. You should be calling the function that has this logic instead. If  you make changes to the function you won't have unit tests that cover that.

Reply:
Thank you.